test/cpp/qps/qps_driver.cc

```diff
@@ -68,49 +82,124 @@ using grpc::testing::ServerType;
 using grpc::testing::RpcType;
 using grpc::testing::ResourceUsage;
 
+using std::vector;
+using std::string;
+
+vector<string> ParseCommaSeparatedIntList(string list) {
+  // Pass by value because strtok consumes its input.
+  vector<string> result;
+  char *p = strtok((char*)list.c_str(), ",");
+  while (p) {
+    result.push_back(p);
+    p = strtok(NULL, ",");
+  }
+  return result;
+}
+
+vector<ClientConfig> GetClientConfigsFromFlags() {
+  vector<ClientConfig> configs;
+  const auto rpc_types = ParseCommaSeparatedIntList(FLAGS_rpc_types);
+  const auto outstanding_rpcs_per_channel =
+      ParseCommaSeparatedIntList(FLAGS_outstanding_rpcs_per_channel);
+  const auto client_channels =
+      ParseCommaSeparatedIntList(FLAGS_client_channels);
+  const auto payload_sizes = ParseCommaSeparatedIntList(FLAGS_payload_sizes);
+  const auto client_types = ParseCommaSeparatedIntList(FLAGS_client_types);
+  const auto async_client_threads =
+      ParseCommaSeparatedIntList(FLAGS_async_client_threads);
+
+  ClientConfig cc;
+  cc.set_enable_ssl(FLAGS_enable_ssl);
+
+  for (const auto& rpc_type_str : rpc_types) {
```

Comment:
Now on to Craig's comment. I agree wholeheartedly. I really shudder at the idea of a 6-nested for-loop that creates protobufs, and it will only get worse as we add other parameters to the list of things to watch for. I would really like to suggest going back to a one-off runner as it has been but control the running options separately in a shell-script, as is done in a simple-form in the heavily underused qps-sweep.sh that lives in this same directory . I understand that that's more moving parts, but the shell is particularly well-suited for the sort of work that you're doing here (supporting range-based for, tokenization, etc since the 1970s). It's trivial to understand that code and add in a new flag as needed. The grpc module should provide the mechanisms and examples but does not need to directly include every use case under the sun, IMO.